DependencyInjection/Configuration.php

```diff
@@ -120,14 +120,18 @@ protected function configurePlugins(ArrayNodeDefinition $root)
             ->arrayNode('plugins')
                 ->addDefaultsIfNotSet()
                 ->children()
-
                     ->arrayNode('authentication')
-                    ->canBeEnabled()
-                        ->children()
-                            ->scalarNode('authentication')
-                                ->info('This must be a service id to a service implementing Http\Message\Authentication')
-                                ->isRequired()
-                                ->cannotBeEmpty()
+                        ->useAttributeAsKey('name')
+                        ->prototype('array')
+                            ->children()
+                                ->enumNode('type')
+                                    ->values(['basic', 'bearer', 'wsse'])
+                                    ->isRequired()
+                                    ->cannotBeEmpty()
+                                ->end()
+                                ->scalarNode('username')->end()
+                                ->scalarNode('password')->end()
+                                ->scalarNode('token')->end()
```

Comment:
we could go all fancy with custom validation to have the right required parameters based on the type. but we don't need that now.

i wonder if we should add type `service` that accepts a scalarNode `authentication_service` to keep supporting custom authentication services?